share/picongpu/tests/compileLaser/include/picongpu/param/incidentField.param

```diff
@@ -0,0 +1,145 @@
+ * parameters
+ *  - profiles::PulseFrontTilt<>      : Gaussian beam with tilted pulse front with given parameters
+ *  - profiles::Wavepacket<>          : wavepacket with Gaussian spatial and temporal envelope profile with given
+ * parameters
+ *
+ * All profiles but `Free<>` and `None` are parametrized with a profile-specific structure.
+ * Their interfaces are defined in the corresponding `.def` files inside directory
+ * picongpu/fields/incidentField/profiles/. Note that all these parameter structures inherit common base structures
+ * from `BaseParam.def`. Thus, a user-provided structure must also define all members according to the base struct.
+ *
+ * In the end, this file needs to define `XMin`, `XMax`, `YMin`, `YMax`, `ZMin`, `ZMax` (the latter two can be skipped
+ * in 2d) type aliases in namespace `picongpu::fields::incidentField`. Each of them could be a single profile or a
+ * typelist of profiles created with `MakeSeq_t`. In case a typelist is used, the resulting field is a sum of
+ * effects of all profiles in the list. This file also has to define constexpr array `POSITION` that controls
+ * positioning of the generating surface relative to total domain. For example:
+ *
+ * @code{.cpp}
+ * using XMin = profiles::Free<UserFunctorIncidentE, UserFunctorIncidentB>;
+ * using XMax = profiles::None;
+ * using YMin = MakeSeq_t<profiles::PlaneWave<UserPlaneWaveParams>, profiles::Wavepacket<UserWavepacketParams>>;
+ * using YMax = profiles::None;
+ * using ZMin = profiles::Polynom<UserPolynomParams>;
+ * using ZMax = profiles::GaussianBeam<UserGaussianBeamParams>;
+ *
+ * constexpr int32_t POSITION[3][2] = { {16, -16}, {16, -16}, {16, -16} };
+ * @endcode
+ */
+
+#pragma once
+
+#include "picongpu/fields/incidentField/profiles/profiles.def"
+
+#ifndef PARAM_A0
+#    define PARAM_A0 8.0
+#endif
+
+#ifndef PARAM_WAVE_LENGTH_SI
+#    define PARAM_WAVE_LENGTH_SI 0.8e-6
+#endif
+
+#ifndef PARAM_PULSE_LENGTH_SI
+#    define PARAM_PULSE_LENGTH_SI 5.e-15
+#endif
+
+namespace picongpu
+{
+    namespace fields
+    {
+        namespace incidentField
+        {
+            /** Functor example to test the incident field Free<> solver implantation.
+             */
+            class ZeroField
+            {
+            public:
+                HINLINE ZeroField(float_X const /*currentStep*/, const float3_64 /*unitField*/)
+                {
+                }
+
+                HDINLINE float3_X operator()(const floatD_X& totalCellIdx) const
+                {
+                    return float3_X::create(0);
+                }
+            };
+
+            using AllLaserProfiles = MakeSeq_t<
+                profiles::DispersiveLaser<>,
+                profiles::ExpRampWithPrepulse<>,
+                profiles::Free<ZeroField, ZeroField>,
+                profiles::GaussianBeam<>,
+                profiles::PlaneWave<>,
+                profiles::Polynom<>,
+                profiles::PulseFrontTilt<>,
+                profiles::Wavepacket<>>;

```

Comment:
Do these profiles do anything without params?

Reply:
Yes each profiles is shipped with default parameters. Therefore the full algorithm is compiled.